app/lib/feed_manager.rb

```diff
@@ -252,18 +253,29 @@ def populate_home(account)
         # because none of its statuses would stay on the feed anyway
         next if last_status_score < oldest_home_score
       end
-
       statuses = target_account.statuses.where(visibility: [:public, :unlisted, :private]).includes(:preloadable_poll, :media_attachments, :account, reblog: :account).limit(limit)
+
       crutches = build_crutches(account.id, statuses)
 
       statuses.each do |status|
         next if filter_from_home?(status, account.id, crutches)
-
         add_to_feed(:home, account.id, status, aggregate_reblogs: aggregate)
       end
 
       trim(:home, account.id)
     end
+
+    TagFollow.where(account_id: account.id).find_each do |target_tag|
+      statuses = target_tag.tag.statuses.where(visibility: [:public, :unlisted, :private]).limit(limit)
```

Comment:
Are we sure this fetches statuses in reverse chronological order?

Reply:
I guess it's better to be sure.  I more or less copied the logic over from the accounts section above.

Reply:
added an explicit sort